homeassistant/components/monzo/config_flow.py

```diff
@@ -0,0 +1,141 @@
+"""Config flow to configure the Monzo component."""
+
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+
+from .const import (DOMAIN, CONF_CLIENT_ID, CONF_CLIENT_SECRET)
+from .local_auth import MonzoAuthCallbackView
+
+
+MONZO_AUTH_START = '/api/monzo'
+MONZO_AUTH_CALLBACK_PATH = '/api/monzo/callback'
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Monzo instances."""
+    return set(
+        '{0}'.format(
+            entry.data[CONF_CLIENT_ID]
+            for entry in hass.config_entries.async_entries(DOMAIN)))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class MonzoFlowHandler(config_entries.ConfigFlow):
+    """Handle an Monzo config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        pass
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        errors = {}
+
+        if user_input is not None:
+            client_id = user_input.get(CONF_CLIENT_ID)
+            client_secret = user_input.get(CONF_CLIENT_SECRET)
+            if None not in (client_id, client_secret):
+                if client_id in configured_instances(self.hass):
+                    errors['base'] = 'identifier_exists'
+                else:
+                    return await self._set_up_redirect(user_input)
+
+        data_schema = OrderedDict()
+        data_schema[vol.Required(CONF_CLIENT_ID)] = str
+        data_schema[vol.Required(CONF_CLIENT_SECRET)] = str
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema(data_schema),
+            errors=errors,
+        )
+
+    async def async_step_link(self, user_input=None):
+        """Attempt to link with the Monzo account.
+
+        Route the user to a website to authenticate with Monzo. Depending on
+        implementation type we expect a pin or an external component to
```

Comment:
If there is a pin method, use that one instead of doing an oauth callback.

Reply:
Unfortunately there isn't, I would have preferred it if so as well.